### PR TITLE
chore: remove legacy copy_to_directory_action

### PR DESCRIPTION
**Pull request by kormide**
### Type of change

- Chore (any other change that doesn't affect source or test files, such as configuration)

**For changes visible to end-users**

- Breaking change (this change will force users to change their own code or config)
- Relevant documentation has been updated
- Suggested release notes are provided below:
- The legacy `copy_to_directory_action` rule helper has been removed. Please use `copy_to_directory_bin_action` action instead.

### Test plan
- Covered by existing test cases

### CODE REVIEW

**Comment by kormide** [comment]
Sorry @gregmagolan, I should have given you the satisfaction of deleting all of this.

File: lib/private/BUILD.bazel [hunk 1]
```diff
@@ -12,3 +14,299 @@ exports_files(
...
+    visibility = ["//lib:__subpackages__"],
+    deps = [":strings"],
+)
+
+bzl_library(
+    name = "copy_common",
+    srcs = ["copy_common.bzl"],
+    visibility = ["//lib:__subpackages__"],
+)
+
```
Comment (kormide) [inline lib/private/BUILD.bazel:30]: I ran gazelle which made a large diff. Perhaps I should have not for this PR.

Comment (alexeagle) [reply lib/private/BUILD.bazel:30]: Yeah gazelle is a bit broken in this repo. If we liked it then we woulda put a CI on it. Likely have to revert some of the incorrect changes it makes.

Comment (kormide) [reply lib/private/BUILD.bazel]: Removed the gazelle diffs.

**Review by alexeagle — APPROVED** [review]

**Pull request merged**